bluesky/callbacks/stream.py

```diff
@@ -0,0 +1,181 @@
+import logging
+import time as ttime
+from collections import Iterable, ChainMap
+
+import streamz
+import jsonschema
+import numpy as np
+from event_model import DocumentNames, schemas
+
+from .core import CallbackBase
+from ..run_engine import Dispatcher
+from ..utils import new_uid
+
+logger = logging.getLogger(__name__)
+
+
+class LiveStream(CallbackBase):
+    """Create a secondary event stream of processed data"""
+    def __init__(self, in_node, out_node=None):
+        self.in_node = in_node
+        # Use the same node as an output if None is given
+        self.out_node = out_node or self.in_node
+        # The output should propagate the modified Event document down the
+        # pipeline
+        self.event = self.in_node.emit
```

Comment:
:+1:

Reply:
what if an event is not filled? should the stream do some checking? (will need a `db` reference passed in as arg/kwarg)